tests/test_model_signature.py

```diff
@@ -0,0 +1,55 @@
+from inspect import signature
+from typing import Any
+
+from pydantic import BaseModel, Extra, Field, create_model
+
+
+def test_model_signature():
+    class Model(BaseModel):
+        a: float = Field(..., title='A')
+        b = Field(10)
+
+    sig = signature(Model)
+    assert sig != signature(BaseModel)
+
+    assert [str(p).replace(' ', '') for p in sig.parameters.values()] == [
```

Comment:
given that you do this a lot, maybe best to implement your own comparison function:

```py
def _eq(a: Union[str, List[str]], b: Union[str, List[str]]) -> bool:
    """
    compare strings with spaces removed
    """
    if isinstance(a, str) and isinstance(b, str):
        return a.replace(' ', '') == b.replace(' ', '')
    elif isinstance(a, list) and isinstance(b, list):
        return all(_eq(a_, b_) for a_, b_ in zip(a, b))
    else:
        raise TypeError(f'arguments must be both strings or both lists, not {type(a)}, {type(b)}')
```